openedx/features/pakx/lms/pakx_admin_app/serializers.py

```diff
@@ -72,13 +74,14 @@ class Meta:
         model = User
         fields = ('id', 'name', 'email', 'last_login', 'assigned_courses', 'incomplete_courses', 'completed_courses')
 
-    def get_assigned_courses(self, obj):
+    @staticmethod
+    def get_assigned_courses(obj):
         return len(obj.enrollments)
 
-    def get_incomplete_courses(self, obj):
-        # todo: placeholder data, use figure's data for course completion once it's integrated
-        return obj.id
+    @staticmethod
+    def get_incomplete_courses(obj):
+        return get_in_progress_courses_count_for_user(obj).count()
 
-    def get_completed_courses(self, obj):
-        # todo: placeholder data, use figure's data for course completion once it's integrated
-        return obj.id
+    @staticmethod
+    def get_completed_courses(obj):
+        return get_completed_courses_count_for_user(obj).count()
```

Comment:
this will run 1 query for every learner, can we update the view's `get_queryset` function by using prefetch so that it won't run query for every user. you can replace `CourseEnrollment` with the model accordingly